api/identifiers/serializers.py

```diff
@@ -45,6 +55,33 @@ def self_url(self, obj):
             },
         )
 
+    def create(self, validated_data):
+        node = self.context['view'].get_node()
+        auth = get_user_auth(self.context['request'])
+        if validated_data.get('category', None) == 'doi':
+            if node.get_identifier('doi'):
+                raise exceptions.ValidationError('A DOI already exists for this resource.')
+            try:
+                identifiers = get_or_create_identifiers(node)
+            except HTTPError:
+                raise exceptions.ValidationError('Error response from client.')
+            except TypeError:
+                raise ServiceUnavailableError()
```

Comment:
v1 view doesn't throw this, but this is for when we're not connected to datacite (like in local development), so we'll throw a 503.